x-pack/plugins/apm/public/components/app/correlations/ml_latency_correlations.tsx

```diff
@@ -0,0 +1,428 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import {
+  EuiIcon,
+  EuiLink,
+  EuiBasicTableColumn,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiProgress,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import { FETCH_STATUS } from '../../../hooks/use_fetcher';
+import {
+  CorrelationsChart,
+  replaceHistogramDotsWithBars,
+} from './correlations_chart';
+import {
+  CorrelationsTable,
+  SelectedSignificantTerm,
+} from './correlations_table';
+import { useCorrelations } from './use_correlations';
+import { createHref, push } from '../../shared/Links/url_helpers';
+import { useUiTracker } from '../../../../../observability/public';
+import { asPreciseDecimal } from '../../../../common/utils/formatters';
+
+const DEFAULT_PERCENTILE_THRESHOLD = 95;
+const isErrorMessage = (arg: unknown): arg is Error => {
+  return arg instanceof Error;
+};
+
+interface Props {
+  onClose: () => void;
+}
+
+interface MlCorrelationsTerms {
+  correlation: number;
+  ksTest: number;
+  fieldName: string;
+  fieldValue: string;
+  duplicatedFields?: string[];
+}
+
+export function MlLatencyCorrelations({ onClose }: Props) {
+  const {
+    core: { notifications },
+  } = useApmPluginContext();
+
+  const { serviceName } = useParams<{ serviceName: string }>();
+  const { urlParams } = useUrlParams();
+
+  const fetchOptions = useMemo(
+    () => ({
+      ...{
+        serviceName,
+        ...urlParams,
+      },
+    }),
+    [serviceName, urlParams]
+  );
+
+  const {
+    error,
+    histograms,
+    percentileThresholdValue,
+    isRunning,
+    progress,
+    startFetch,
+    cancelFetch,
+    overallHistogram: originalOverallHistogram,
+  } = useCorrelations({
+    index: 'apm-*',
+    ...{
+      ...fetchOptions,
+      percentileThreshold: DEFAULT_PERCENTILE_THRESHOLD,
+    },
+  });
+
+  const overallHistogram = useMemo(
+    () => replaceHistogramDotsWithBars(originalOverallHistogram),
+    [originalOverallHistogram]
+  );
+
+  // cancel any running async partial request when unmounting the component
+  // we want this effect to execute exactly once after the component mounts
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => cancelFetch, []);
+
+  // start fetching on load
+  // we want this effect to execute exactly once after the component mounts
+  useEffect(() => {
+    startFetch();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
did you try:
```suggestion
  useEffect(() => {
    startFetch();
    
    return cancelFetch;
  }, []);
```

Reply:
Updated here https://github.com/elastic/kibana/pull/99905/commits/fdeafd869ab95e17cc81598d1633a646415e9f38